File: model/Event.cpp
//
// Created by Lucas Kujawski on 27/12/2020.
//

#include "Event.h"

Event::Event(int timestamp, char *category, char *action, char *label, Client *client) : timestamp(timestamp),
                                                                                         category(category),
                                                                                         action(action), label(label),
                                                                                         client(client) {}

File: model/event.py
import datetime

from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy import ForeignKey
from sqlalchemy.orm import relationship

from database import T4U_BASE


class Event(T4U_BASE):

    __tablename__ = 'event'
    id = Column(Integer, primary_key=True)
    timestamp = Column(DateTime, default=datetime.datetime.utcnow)
    category = Column(String(255))
    action = Column(String(255))
    label = Column(String(255))
    client_id = Column(Integer, ForeignKey('user.id'))
    client = relationship("Client")

    def __repr__(self):
        return "[%s] %s|%s|%s" % (self.timestamp, self.category, self.action, self.label)

    def get_full_description(self):
        return "Date: %s\nCategory: %s\nAction: %s\nLabel: %s\nClient: %s" % (self.timestamp.strftime("%B %d, %Y"),
                                                                              self.category, self.action, self.label,
                                                                              self.client)